src/shared/clients/defaultStsClient.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { STS } from 'aws-sdk'
+import { ServiceConfigurationOptions } from 'aws-sdk/lib/service'
+import { ext } from '../extensionGlobals'
+import { StsClient } from './stsClient'
+
+export class DefaultStsClient implements StsClient {
+
+    private readonly _credentials: ServiceConfigurationOptions | undefined
+
+    public constructor (
+        public readonly regionCode: string,
+        credentials?: { accessKeyId: string, secretAccessKey: string }
```

Comment:
Credentials are an edge case--typically we're using the profile. Credentials are specifically for supporting immediate credential validation when adding new credentials through VS Code prompts.

I think I want to add multiple constructors for this, specifically, to handle the following cases:
```
DefaultStsClient()
DefaultStsClient(regionCode)
DefaultStsClient(credentials)
DefaultStsClient(regionCode, credentials)
```

DefaultStsClient doesn't need a region for getCallerIdentity, and this prevents us from needing a default `us-east-1` declaration. I don't want to roll the two into a `DefaultStsClientArgs` interface because this falls out of line with the other client constructors.

Reply:
What's the advantage of the multiple constructors over optional parameters?

```typescript
args: {
    regionCode?: string
    credentials?: ...
}
```